### PR TITLE
Initial Hugo resource

### PR DESCRIPTION
**Pull request by pasimko**

### CODE REVIEW

File: resources/osl_hugo.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+action :install do
+  package 'tar'
+
+  ark 'hugo' do
+    url "https://github.com/gohugoio/hugo/releases/download/v#{new_resource.version}/hugo_#{new_resource.version}_Linux-64bit.tar.gz"
+    prefix_root '/opt'
+    prefix_home '/opt'
+    strip_components 0
+    version new_resource.version
+  end
```
Comment (ramereth) [inline resources/osl_hugo.rb:18]: Please add a symlink `/usr/local/bin/hugo` to `/opt/hugo/hugo`

File: test/integration/osl_hugo/controls/osl_hugo.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+control 'osl_hugo' do
+  describe directory('/opt/hugo-0.120.4') do
+    it { should exist }
+  end
+  describe directory('/opt/hugo') do
+    it { should exist }
+    its('link_path') { should cmp '/opt/hugo-0.120.4' }
+  end
```
Comment (ramereth) [inline test/integration/osl_hugo/controls/osl_hugo.rb:8]: Let's replace these tests with a command test with `hugo version` returning the version you expect with an exit status of 0

File: test/integration/osl_hugo/controls/osl_hugo.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  describe file('/opt/hugo/hugo') do
+    it { should exist }
```
Comment (ramereth) [inline test/integration/osl_hugo/controls/osl_hugo.rb:10]: Also add the size is greater than zero.

File: test/integration/osl_hugo/controls/osl_hugo.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    its('mode') { should cmp '0755' }
+  end
```
Comment (ramereth) [inline test/integration/osl_hugo/controls/osl_hugo.rb:12]: Add a test for the symlink.

**Review by ramereth — CHANGES_REQUESTED** [review]
Great start! Don't forget to add spec tests.

File: kitchen.yml [hunk 5]
```diff
@@ -126,6 +126,12 @@ suites:
     provisioner:
       enforce_idempotency: false
       multiple_converge: 1
+  - name: osl_hugo
+    run_list:
+      - recipe[osl-resources-test::osl_hugo]
+    provisioner:
+      enforce_idempotency: false
+      multiple_converge: 1
```
[the remark this answers is not in the document]
Comment (ramereth) [reply kitchen.yml:131]: Please fix this

File: kitchen.yml [hunk 6]
```diff
@@ -126,6 +123,12 @@ suites:
     provisioner:
       enforce_idempotency: false
       multiple_converge: 1
+  - name: osl_hugo
+    run_list:
+      - recipe[osl-resources-test::osl_hugo]
+    provisioner:
+      enforce_idempotency: true
+      multiple_converge: 2
```
Comment (ramereth) [inline kitchen.yml:131]: Please remove this, it will pull this from the global config above.

**Review by ramereth — APPROVED** [review]
Minor bump

**Comment by pasimko** [comment]
!bump minor ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `main` and has automatically performed a minor-level version bump to v1.8.0.  Have a nice day!

**Pull request merged**